client/src/components/TimetableSwitcher/TimetableSwitcher.tsx

```diff
@@ -58,6 +64,25 @@ const TimetableSwitcher = memo(() => {
 		else setAction(inputAction);
 	};
 
+    const handleSaveTimetable = async () => {
+        const input = document.getElementById('timetable') as HTMLElement;
+        const imgData = await domtoimage.toPng(input)
+        triggerBase64Download(imgData, 'FFCSThingy - Timetable');
+    };
+
+    const handleSaveCourses = async () => {
+        let csv = '';
+        let header = Object.keys(timetable[0]).filter((val, index) => index!==0).join(',');
```

Comment:
This might fail if `timetable` is `[]`. Add nullish checks please.
Also run `npm lint` or add `ESLint` as an extension to your editor. 
```suggestion
        let header = Object.keys(timetable[0]).filter((val, index) => index !== 0).join(',');
```